### PR TITLE
feat(engine): Make KRE Jetstream compatible

### PR DESCRIPTION
**Pull request by alexrodfe**
## Why
As our Runners are now Jetstream compatible, we need KRE to be Jetstream compatible as well .

## What
- K8s manager now sets new environment variables regarding stream naming and new node setup.
- K8s manager will delete workflow's streams on version deletion to free up memory.
- There has been also some refactor and renaming

### CODE REVIEW

File: engine/k8s-manager/kubernetes/version/nats.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package version
```
[the remark this answers is not in the document]
Comment (alejandrogr) [reply engine/k8s-manager/kubernetes/version/nats.go:1]: Yes, I'm with @JYisus we should separate all the NATs code to a separate package

**Review by alejandrogr — CHANGES_REQUESTED** [review]
This is a major release, we should rise the major version in the github workflow.

File: engine/k8s-manager/kubernetes/version/nats.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexrodfe) [reply engine/k8s-manager/kubernetes/version/nats.go]: Done :+1:

File: helm/kre/templates/engine/nats-streaming/statefulset.yaml [hunk 2]
```diff
@@ -44,6 +54,4 @@ spec:
         storageClassName: {{ .Values.nats_streaming.storage.className }}
         resources:
           requests:
-            storage: {{ .Values.nats_streaming.storage.size }}
-
-
+            storage: {{ .Values.nats_streaming.storage.size }}
```
[the remark this answers is not in the document]
Comment (alexrodfe) [reply helm/kre/templates/engine/nats-streaming/statefulset.yaml]: Done!

**Review by JYisus — APPROVED** [review]

**Review by alejandrogr — APPROVED** [review]
LGTM! Welcome Jetstream! :D

**Pull request merged**